pkgs/tools/typesetting/tex/texlive/build-tex-env.nix

```diff
@@ -87,14 +130,45 @@ let
     ];
   };
 
-in (buildEnv {
+  meta = {
+    description = "TeX Live environment";
+    platforms = lib.platforms.all;
+    outputsToInstall = [ "out" ]
+      ++ lib.optional (splitOutputs ? man) [ "man" ];
+  };
+
+  # emulate split output derivation
+  splitOutputs = {
+    out = out // { outputSpecified = true; };
+    texmfdist = texmfdist // { outputSpecified = true; };
+    texmfroot = texmfroot // { outputSpecified = true; };
+  } // (lib.genAttrs pkgList.nonEnvOutputs (outName: buildEnv {
+    name = name + "-" + outName;
+    paths = builtins.catAttrs "outPath"
+      (pkgList.otherOutputs.${outName} or [ ] ++ pkgList.specifiedOutputs.${outName} or [ ]);
+    inherit passthru;
+  } // { outputSpecified = true; }));
+
+  passthru = splitOutputs // {
+    all = builtins.attrValues splitOutputs;
+    outputs = [ "out" ] ++ pkgList.nonEnvOutputs;
+    # This is set primarily to help find-tarballs.nix to do its job
+    requiredTeXPackages = lib.filter lib.isDerivation pkgList.all;
+    # useful for inclusion in the `fonts.packages` nixos option or for use in devshells
+    fonts = "${texmfroot}/texmf-dist/fonts";
+    overrideTeXConfig = newArgs: self (args // { __calledFromCombine = false; } // newArgs);
+    withPackages = reqs: self (args // { requiredTeXPackages = ps: requiredTeXPackages ps ++ reqs ps; __calledFromCombine = false; });
+  };
```

Comment:
I have decided that fake multi-output is better here, where all TeX-relevant outputs are captured in `texmfdist` and the likes, and the rest are combined together in separate outputs.

99% of the time, it will result in additional `man` and `info` outputs that can be installed on demand, but will be ignored when using texlive as build input. Why do that: too many things can happen when a user can pass a non-texlive package (e.g. what happens to its man pages?), but with this approach, the result is ver predictable.

Apart from that, the main feature here is `.overrideTeXConfig`. This is essentially equivalent to a normal `.override` from `makeOverridable`, but with a different name to distinguish it from `texlive.override` (which changes the inputs).